### PR TITLE
Update builds and tests

### PR DESCRIPTION
**Pull request by alerickson**
AppVeyor yaml test

### CODE REVIEW

**Comment by jianyunt** [comment]
Please update PR title to meaningful way. Thanks.

File: Test/ModuleTests/tests/legacy.tests.ps1 [hunk 1]
```diff
@@ -2,7 +2,7 @@ $IsLegacyTestRun = (Get-Variable -Name IsLegacyTestRun -ErrorAction Ignore) -and
 
 Describe "Legacy tests" -Tags "Legacy" {
     It "Can import on legacy PowerShell Core" -Skip:((-not $IsLegacyTestRun)) {
-        try { Import-Module PackageManagement -RequiredVersion 1.2.1 } catch {}
+        try { Import-Module PackageManagement -RequiredVersion 1.2.1} catch {}
```
Comment (jianyunt) [inline Test/ModuleTests/tests/legacy.tests.ps1]: This test is failing now;If we no longer support the case where the current version of PackageManagement run on previous pscore, then we can drop off this test case.

 The impact may be nano server. But we do not know whether they are actually upgrade packageManagement.  The other scenario may be MicrosoftDockerProvider. You may ask them about our new requirement.

**Comment by bmanikm** [comment]
@alerickson Please update the title of this PR.

File: Test/ModuleTests/tests/install-packageprovider.tests.ps1 [hunk 2]
```diff
@@ -33,15 +33,15 @@ try {
     catch { }
 }
 
-$InternalGallery = "https://dtlgalleryint.cloudapp.net/api/v2/"
-$InternalGallery2 = "http://dtlgalleryint.cloudapp.net/api/v2/"
+$InternalGallery = "https://www.poshtestgallery.com/api/v2/"
+$InternalGallery2 = "https://www.powershellgallery.com/api/v2/"
```
[the remark this answers is not in the document]
Comment (alerickson) [reply Test/ModuleTests/tests/install-packageprovider.tests.ps1]: Okay, I'll change that

**Pull request closed without merging**